src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -624,6 +624,59 @@ private static string UnescapeDriveColon(string fileUri)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Converts a file system path into a DocumentUri required by Language Server Protocol.
+        /// </summary>
+        /// <remarks>
+        /// When sending a document path to a LSP client, the path must be provided as a
+        /// DocumentUri in order to features like the Problems window or peek definition
+        /// to be able to open the specified file.
+        /// </remarks>
+        /// <param name="path">
+        /// A file system path. Note: if the path is already a DocumentUri, it will be returned unmodified.
+        /// </param>
+        /// <returns>The file system path encoded as a DocumentUri.</returns>
+        internal static string ConvertPathToDocumentUri(string path)
+        {
+            const string fileUriPrefix = "file:///";
+
+            if (path.StartsWith("untitled:", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (path.StartsWith(fileUriPrefix, StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return new Uri(path).AbsoluteUri;
+            }
+
+            // VSCode file URIs on Windows need the drive letter lowercase, and the colon
+            // URI encoded. System.Uri won't do that, so we manually create the URI.
+            var newUri = System.Web.HttpUtility.UrlPathEncode(path);
+            int colonIndex = path.IndexOf(":", StringComparison.Ordinal);
```

Comment:
Would this work here instead?

```suggestion
            int colonIndex = path.IndexOf(':');
```

Reply:
Do you think that will shut up the Codacy locale warning?

Reply:
If not, in this instance it may be better to ignore or suppress it.  I'm guessing the method for finding the index of an entire string is a lot more complicated than the one for a single character.  It'd be nice if there was an equivalent culture specific overload though :/

Reply:
Is one a LINQ-provided method? I've seen ones like `Contains()` where LINQ provides the `char` overload, and I wonder if the `string`-specific version is possibly faster. It probably uses [Boyer-Moore](https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore_string-search_algorithm), which for a longer string would be faster than sequential search, but here probably degenerates to "about as fast as a char search" but with locale settings possible.

Reply:
Nah, LINQ doesn't usually deal with indexes since `IEnumerable<>` doesn't implement an indexer.  Closest they have is `ElementAt`.

`string.IndexOf(char)` goes right to `string.IndexOf(char, int, int)` which is an internal call

`string.IndexOf(string, StringComparison)` goes through a few layers of checks and methods before hitting `CompareInfo.InternalFindNLSStringEx` which is a QCall.

The first almost definitely has better performance, but that's not really my concern.  I think that warning only makes sense if being more explicit doesn't drastically change the code path.  Also maybe it's just me but `char` seems to have a much heavier implication of ordinal compare.

Neither will be a measurable change though, so nbd ya'll disagree.

Reply:
(Looks like .NET [doesn't use Boyer-Moore in this case](https://github.com/dotnet/corefx/blob/5e3fe25d1236cd0192dd966f7286fe008d5cf875/src/Common/src/CoreLib/System/Globalization/CompareInfo.cs#L545-L591), but it does look fairly tuned at least)

Reply:
Trying the signature that takes a char. We'll see if Codacy complains.